Dockerfile

```diff
@@ -1,7 +1,8 @@
 FROM qmkfm/qmk_cli
-# test layout ADD layout_src ./layout_src
-# TODO would be nice to make this whole action more reusable parameterising `qmk setup`
-RUN qmk setup zsa/qmk_firmware -b firmware20 -y
+ARG FORK=zsa/qmk_firmware
+ARG BRANCH
```

Comment:
Why no default for BRANCH? I think that breaks backwards compatibility

Reply:
> Why no default for BRANCH? I think that breaks backwards compatibility

because empty branch means always latest branch from zsa/qmk_firmware 
that's the whole point to always get the latest firmware which is firmware21 atm instead of having to manually update it every time. If you don't want the latest firmware you can still set it in the input vars

Reply:
Not entirely correct... 
https://github.com/Phoscur/oryx-macro-hax/actions/runs/3255148167/jobs/5344161975
> Phoscur/QMK-compile/main/action.yml (Line: 38, Col: 9): Unexpected value 'FORK=zsa/qmk_firmware BRANCH='
Error: The template is not valid. Phoscur/QMK-compile/main/action.yml (Line: 38, Col: 9): Unexpected value 'FORK=zsa/qmk_firmware BRANCH='
Error: The CancellationTokenSource has been disposed.

Reply:
Seems I am too stupid right now:
> Error: The template is not valid. Phoscur/QMK-compile/main/action.yml (Line: 38, Col: 9): Unexpected value 'FORK=zsa/qmk_firmware BRANCH={"":-"latest"}',System.FormatException: Input string was not in a correct format.

Reply:
This one is not the issue
> RUN git clone ${BRANCH:+-b $BRANCH --single-branch} [...]
it will just checkout all branches.

With my latest try it still does not work:
> Error: Phoscur/QMK-compile/main/action.yml (Line: 39, Col: 9): Unexpected value 'FORK=zsa/qmk_firmware BRANCH=firmware21'

Reply:
Okay found it now, your syntax using `=` intead of `:` was wrong, also syntax issues in the docker-compose file... @5andr0 would you please have another look and fix it?
> Run docker compose build
no configuration file provided: not found
Error: Process completed with exit code 14.

Reply:
There is an issue with the checkout action, it checks out for the current workflow, not the action repo.

Reply:
Ok fixed that too, now it runs again but does not export artifacts. 

@5andr0 did you attempt to test this or did you think that was my job?